mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp

```diff
@@ -299,6 +299,10 @@ convertScalarType(const spirv::TargetEnv &targetEnv,
 /// supported integer types.
 static Type convertSubByteIntegerType(const SPIRVConversionOptions &options,
                                       IntegerType type) {
+  if (type.getWidth() > 8) {
+    LLVM_DEBUG(llvm::dbgs() << "not a subbyte type\n");
+    return nullptr;
+  }
```

Comment:
This feels like a strange thing to have a runtime error for; if it's for convenience, maybe it should be documented (that nothing is returned if it's not sub-byte), but otherwise it feels like something you'd use an `assert` for, as it represents a bug if this is called on a non-sub-byte type? Maybe others would disagree with me.

Reply:
Added a comment. Retuning null on unsupported types follows the general stye of the type conversion in mlir.